File: core/core-domain/src/main/java/com/nmh/commerce/coupon/MinimumPriceConstraintPolicy.java
package com.nmh.commerce.coupon;

import com.nmh.commerce.domain.Money;
import com.nmh.commerce.product.Product;
import lombok.Getter;

@Getter
public class MinimumPriceConstraintPolicy implements ConstraintPolicy {
    private final Money minimumPrice;

    public MinimumPriceConstraintPolicy(Money minimumPrice) {
        this.minimumPrice = minimumPrice;
    }

    @Override
    public void verify(Product product) {
        if (product.getPrice().isLessThan(minimumPrice)) {
            throw new IllegalArgumentException("쿠폰을 적용 할 수 있는 최소 금액 미달입니다.");
        }
    }
}

File: core/core-domain/src/main/kotlin/com/nmh/commerce/coupon/MinimumPriceConstraintPolicy.kt
package com.nmh.commerce.coupon

import com.nmh.commerce.domain.Money
import com.nmh.commerce.product.Product

class MinimumPriceConstraintPolicy(
    val minimumPrice: Money
) : ConstraintPolicy {
    override fun verify(product: Product) {
        require(!product.price.isLessThan(minimumPrice)) { "쿠폰을 적용 할 수 있는 최소 금액 미달입니다." }
    }
}
